src/GraphQL.Tests/Complexity/ComplexityMetaDataTests.cs

```diff
@@ -0,0 +1,149 @@
+        result.TotalQueryDepth.ShouldBe(4);
+    }
+
+    [Fact]
+    public async Task MetaData_With_Fragments_ShouldBe_Used()
+    {
+        var result = await _fixture.AnalyzeAsync(@"
+query {
+    hero { #2
+        id #0
+        name #2
+        ...friendsRoot #132(4/2*66)
+    }
+}
+
+fragment friendsRoot on Hero {
+    friends { #3
+        ...friendsIdName #27(9/2*6)
+        f1: friends { #9
+            name #9
+        }
+        f2: friends { #9
+            name #9
+        }
+    }
+}
+
+fragment friendsIdName on Hero {
+    friends { #3
+        id #0
+        name #3
+    }
+}
+").ConfigureAwait(false);
+
+        result.Complexity.ShouldBe(136);
+        result.TotalQueryDepth.ShouldBe(5);
+    }
+}
+
+public class ComplexityMetaDataFixture : IDisposable
+{
+    public class ComplexitySchema : Schema
+    {
+        public ComplexitySchema()
+        {
+            Query = new ComplexityQuery();
+        }
+    }
+
+    public class ComplexityQuery : ObjectGraphType
+    {
+        public ComplexityQuery()
+        {
+            Field<Hero>("hero", resolve: _ => 0)
+                .WithImpact(2);
+        }
+    }
+
+    public class Hero : ObjectGraphType<int>
+    {
+        public Hero()
+        {
+            Field<IntGraphType>("id", resolve: context => context.Source)
+                .WithImpact(0);
+
+            Field<StringGraphType>("name", resolve: context => $"Tom_{context.Source}");
+
+            Field<ListGraphType<Hero>>("friends", resolve: context => new List<int> { 0, 1, 2 }.Where(x => x != context.Source).ToList())
+                .WithImpact(3);
+        }
+    }
+
+    private readonly ServiceProvider _provider;
+    private readonly ComplexityAnalyzer _analyzer;
+    private readonly IDocumentBuilder _documentBuilder;
+    private readonly ComplexityConfiguration _config;
+    private readonly ISchema _schema;
+    private readonly IDocumentExecuter _executer;
+
+    public ComplexityMetaDataFixture()
+    {
+        _provider = new ServiceCollection()
+            .AddGraphQL(builder => builder
+                .AddSchema<ComplexitySchema>()
+                .AddComplexityAnalyzer()
+            ).BuildServiceProvider();
+
+        _documentBuilder = _provider.GetRequiredService<IDocumentBuilder>();
+        _config = new();
+        _schema = _provider.GetRequiredService<ISchema>();
+        _analyzer = (ComplexityAnalyzer)_provider.GetRequiredService<IComplexityAnalyzer>();
+        _executer = _provider.GetRequiredService<IDocumentExecuter<ComplexitySchema>>();
+    }
+
+    public async Task<ComplexityResult> AnalyzeAsync(string query)
+    {
+        var result = await _executer.ExecuteAsync(o =>
+        {
+            o.Query = query;
+            o.RequestServices = _provider;
+        }).ConfigureAwait(false);
+        result.Errors.ShouldBeNull();
+
+        return _analyzer.Analyze(_documentBuilder.Build(query), _config.FieldImpact ?? 2f, _config.MaxRecursionCount, _schema);
```

Comment:
```suggestion
        return _analyzer.Validate(_documentBuilder.Build(query), _config, _schema);
```

Reply:
`Analyze` and `Validate` methods don't have the same return signature, the last returns `void`.

Reply:
ok